src/incidentFeed.ts

```diff
@@ -0,0 +1,128 @@
+import settings from 'electron-settings';
+import { EventEmitter } from 'events';
+import Parser from 'rss-parser';
+
+const FEED_URL = 'https://www.netlifystatus.com/history.rss';
+
+interface IFeedItem {
+  title: string;
+  pubDate: string;
+  content: string;
+  link: string;
+}
+
+interface IConnection {
+  isOnline: boolean;
+}
+
+export default class IncidentFeed extends EventEmitter {
+  private parser: { parseURL(feedUrl: string) };
+  private feed: IFeedItem[];
+
+  constructor(connection: IConnection) {
+    super();
+    this.parser = new Parser();
+    this.feed = [];
+    let isInitialFetch = true;
+
+    // main update loop for checking Netlify Incidents RSS
+    const repeat = () => {
+      setTimeout(
+        async () => {
+          if (connection.isOnline) {
+            const fetchedFeed: IFeedItem[] = await this.fetchAndParseFeed();
+            // on the first fetch, kee notify about any very recent updates...
+            if (isInitialFetch) {
+              this.feed = fetchedFeed;
+              const recentIncidents = this.feed.filter(item => {
+                const publicationDate = new Date(item.pubDate);
+                return isToday(publicationDate) || isYesterday(publicationDate);
+              });
+              recentIncidents.forEach(item => {
+                this.notify({
+                  body: item.title,
+                  link: item.link,
+                  title: 'Recently reported incident'
+                });
+              });
+              isInitialFetch = false;
+            } else {
+              // on subsequestion fetches, notify on new or updated incidents
+              const newItems = this.findNewItems(fetchedFeed);
+              const updatedItems = this.findUpdatedItems(fetchedFeed);
+              newItems.forEach(item => {
+                this.notify({
+                  body: item.title,
+                  link: item.link,
+                  title: 'New incident reported'
+                });
+              });
+              updatedItems.forEach(item => {
+                this.notify({
+                  body: item.title,
+                  link: item.link,
+                  title: 'Incident updated'
+                });
+              });
+              this.feed = fetchedFeed;
+            }
+          }
+          repeat();
+        },
+        // recheck connection every second, then poll every minute after initial fetch
+        isInitialFetch ? 1000 : 60000
+      );
+    };
+    repeat();
+  }
+
+  public getFeed(): ReadonlyArray<IFeedItem> {
+    return this.feed as ReadonlyArray<IFeedItem>;
+  }
+
+  private notify({ body, title, link }): void {
+    if (settings.get('showNotifications')) {
+      const notification = new Notification({
+        body,
+        title
+      });
+
+      notification.on('click', () => {
+        shell.openExternal(link);
+      });
+
+      // notifications with an attached click handler
```

Comment:
yeah - I was thinking it would be good to have a notifications.js, but I decided to hold off on any refactoring in menubar.js. I'll see what I can do 🙌

Reply:
That's fair. :) As you prefer we could also leave it as is and create another issue for the notification handling. I have no strong feelings here. :)

What I would like to include are tests, though. And the feed class is way harder to test when it includes notification handling (and the repeating). :)

Reply:
Notifications refactored into notify.js, and tested. Mocking out electron kicked my butt -https://github.com/electron/electron/issues/3909

https://github.com/stefanjudis/netlify-menubar/blob/f324d2a407ddeec9fcdd546a59300e88d648be38/src/notify.ts

https://github.com/stefanjudis/netlify-menubar/blob/f324d2a407ddeec9fcdd546a59300e88d648be38/src/notify.test.ts

Reply:
Did you try [this approach](https://github.com/stefanjudis/netlify-menubar/blob/master/src/netlify.test.ts#L8-L16)? I struggled initially, too, but found this okay'ish.

Reply:
Hmmm - nope, I missed that! I just  looked for imports from “electron” in the existing files and when I didn’t see any I figured it wasn’t being mocked anywhere.

does that jest.mock(‘electron’) replace the electron module imported in netlify.ts?

Reply:
> does that jest.mock(‘electron’) replace the electron module imported in netlify.ts?

Yes. This way you can control what will be required. DO you think that would work for you?